### ISSUE TITLE
The application may be doing too much work on its main thread.

### ISSUE THREAD

**Issue by adityamv7** [flutter/flutter#11533]
## Steps to Reproduce
Create an demo app using git cmd run it android emulator 

## Logs
```
Launching lib\main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
Built build\app/outputs/apk/app-debug.apk (21.3MB).
Installing build\app/outputs/apk/app.apk...
Syncing files to device Android SDK built for x86...
I/Choreographer( 2414): Skipped 169 frames!  The application may be doing too much work on its main thread.
```
## Flutter Doctor
```
G:\flutter>flutter doctor
[√] Flutter (on Microsoft Windows [Version 10.0.14393], locale en-IN, channel alpha)
    • Flutter at G:\flutter
    • Framework revision bbcfb8d51a (5 days ago), 2017-08-01 14:50:29 -0700
    • Engine revision f2af347363
    • Tools Dart version 1.25.0-dev.7.0

[√] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at G:\sdk
    • Platform android-25, build-tools 25.0.3
    • ANDROID_HOME = G:\sdk
    • Java binary at: G:\Android\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_76-release-b03)

[√] Android Studio (version 2.2)
    • Android Studio at G:\Android
    • Java version OpenJDK Runtime Environment (build 1.8.0_76-release-b03)

[√] IntelliJ IDEA Community Edition (version 2017.2)
    • Flutter plugin version 16.0
    • Dart plugin version 172.3544.34

[√] Connected devices
    • None
```

**Comment by adityamv7** [comment]
@eseidelGoogle  any specific reason as why this error is happening

**Comment by parth181195** [comment]
i am also getting this message with in my moto g 3 turbo

**Comment by Hixie** [comment]
Does this only happen initially, on load, or does it happen subsequently after that?

**Comment by parth181195** [comment]
only initially on first startup

**Comment by parth181195** [comment]
i think its device related cause i saw many people getting this message on moto g3

**Comment by abinvp** [comment]
I am also getting this message even for a simple "Hello World" application. Seeing in both emulator and galaxy note 8.

**Comment by eseidelGoogle** [comment]
@abinvp that's definitely unexpected, especially on a device.  Do you only see it during startup, or during another time while using the app?

**Comment by zoechi** [comment]
Is this only in debug mode or also in release mode?

**Comment by adityamv7** [comment]
@zoechi its in debug mode

**Comment by eyoeldefare** [comment]
I am also experiencing this

**Comment by DhruvamSharma** [comment]
My app just said: 
Skipped 654 frames!  The application may be doing too much work on its main thread.

And it's a really simple list app.

**Comment by zoechi** [comment]
@DhruvamSharma is this a debug or a profile/release build?
Performance measure doesn't make sense in debug builds because it's runtime behavior is entirely different (JIT/AOT and others)

**Comment by DhruvamSharma** [comment]
@zoechi  It's a debug build. That's for sure. 
Can you help me out with the current situation?

**Comment by zoechi** [comment]
What situation? If it's a debug build it's not relevant. Use profile builds.
See also https://flutter.io/docs/testing/ui-performance

**Comment by DhruvamSharma** [comment]
Thanks 👍

**Comment by keshava19** [comment]
I got the same issue for release build and the application was just a simple application using canvas to draw lines and this happened after the flutter 1.2.1, before upgrade the application didn't have any issues.

**Comment by eseidelGoogle** [comment]
It would help us if someone could either post to this bug (or file a new bug) with more information, including:
-  What example/application demonstrates this (if it's just a stock `flutter create` that's even better).
- Run with what flutter command (e.g. --debug vs. --release)
- On what device/hardware/OS version.
- Using what Flutter version (`flutter doctor` output).

Until we can reproduce this locally, it's going to be hard to get an engineer to fix it. :/

**Comment by keshava19** [comment]
- Application in which I found issue - [Draw](https://github.com/kekahr/draw)
- Both `--debug` or `--release` command is reproducing the issue
- Hardware device, Moto G4 running 7.0 (Nougat), Security Patch 1st April 2018, NonRooted
- Flutter doctor :

    [√] Flutter (Channel stable, v1.2.1, on Microsoft Windows [Version 10.0.17134.590], locale en-US)
        • Flutter version 1.2.1 at C:\src\flutter
        • Framework revision 8661d8aecd (3 weeks ago), 2019-02-14 19:19:53 -0800
        • Engine revision 3757390fa4
        • Dart version 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

    [√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
        • Android SDK at C:\Users\---\AppData\Local\Android\Sdk
        • Android NDK location not configured (optional; useful for native profiling support)
        • Platform android-28, build-tools 28.0.3
        • ANDROID_HOME = C:\Users\---\AppData\Local\Android\Sdk
        • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
        • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
        • All Android licenses accepted.

    [√] Android Studio (version 3.2)
        • Android Studio at C:\Program Files\Android\Android Studio
        • Flutter plugin version 29.1.1
        • Dart plugin version 181.5656
        • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

    [√] IntelliJ IDEA Community Edition (version 2018.2)
        • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2.5
        • Flutter plugin version 31.3.3
        • Dart plugin version 182.5124

    [√] VS Code, 64-bit edition (version 1.25.1)
        • VS Code at C:\Program Files\Microsoft VS Code
        • Flutter extension version 2.21.1

    [√] Connected device (1 available)
        • Moto G 4 • ZY223HDMWR • android-arm • Android 7.0 (API 24)

**Comment by eseidelGoogle** [comment]
That's extremely helpful @keshava19, thank you.

**Comment by rowild** [comment]
Would just like to report that I just run into the same error message, when trying t rebuild a Google demo: 
https://www.youtube.com/watch?v=kn0EOS-ZiIc&t=19s
https://gist.github.com/efortuna/4d56af7586d5cbe0d98bc3d5f9ef5243

Android Emulator runs a Pixel 3 Api 23:5554

It happens on startup ('... dropped 64 frames'), then the app runs normal.

**Comment by meshkani75** [comment]
I git this error in android emulator pixel 2 API 28. `flutter clean` and restarting emulator helped me to solve the problem.

**Comment by legut2** [comment]
Any update on this?

**Comment by Hixie** [comment]
We haven't had a chance to look at this yet. (It shouldn't cause any user-visible problems, because Android's main thread is not the same as flutter's main thread.)

@liyuqian you may wish to add this to your list of performance bugs you are looking at next year.

**Comment by liyuqian** [comment]
Adding it to my list with a conservative milestone of Q1 next year.

**Comment by SwissCheese5** [comment]
Yesterday, my pixel 4a emulator spammed the message while my app was idle. Nothing wrong with it, performance was flawless, but the console kept spamming the message.

**Comment by mingsai** [comment]
I can see the same error happening in many stubs running in debug mode that I am just ignoring. I am seeing them on device and in the emulators. Many of my github/gists have additional samples displaying the behavior. 

When it's time to reproduce this just look to your own performance code. I've snipped a copy here (https://gist.github.com/mingsai/7c1e8dba62d89486cfdc8817a9fe9065) also see my github profile for others. Rename the MyApp to MaterialApp on the tests. Running in a Pixel 3a XL emulator, Android 10,12 GB on disk. I can see this  error happening intermittently in the log:

Launching lib/main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...
✓ Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...
D/HostConnection( 5268): createUnique: call
D/HostConnection( 5268): HostConnection::get() New Host Connection established 0xc5a37320, tid 5302
D/HostConnection( 5268): createUnique: call
D/HostConnection( 5268): HostConnection::get() New Host Connection established 0xd98818a0, tid 5294
D/HostConnection( 5268): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV420_888_to_NV21 ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_gles_max_version_3_0 
D/eglCodecCommon( 5268): allocate: Ask for block of size 0x1000
D/eglCodecCommon( 5268): allocate: ioctl allocate returned offset 0x3ffff6000 size 0x2000
D/HostConnection( 5268): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV420_888_to_NV21 ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_gles_max_version_3_0 
D/EGL_emulation( 5268): eglMakeCurrent: 0xf017fa60: ver 3 0 (tinfo 0xe4eeb1c0)
I/_thread_exampl( 5268): NativeAlloc concurrent copying GC freed 1855(186KB) AllocSpace objects, 3(60KB) LOS objects, 57% free, 1134KB/2670KB, paused 30us total 607.221ms
Syncing files to device Android SDK built for x86...
D/EGL_emulation( 5268): eglMakeCurrent: 0xdcd1a0c0: ver 3 0 (tinfo 0xc0da4cb0)
I/Choreographer( 5268): Skipped 193 frames!  The application may be doing too much work on its main thread.
D/EGL_emulation( 5268): eglMakeCurrent: 0xf017fa60: ver 3 0 (tinfo 0xe4eeb1c0)
Syncing files to device Android SDK built for x86...
W/_thread_exampl( 5268): JNI critical lock held for 19.477ms on Thread[1,tid=5268,Runnable,Thread*=0xf0137800,peer=0x720ac1f0,"main"]
W/_thread_exampl( 5268): JNI critical lock held for 21.297ms on Thread[1,tid=5268,Runnable,Thread*=0xf0137800,peer=0x720ac1f0,"main"]
W/_thread_exampl( 5268): JNI critical lock held for 16.821ms on Thread[1,tid=5268,Runnable,Thread*=0xf0137800,peer=0x720ac1f0,"main"]
W/_thread_exampl( 5268): JNI critical lock held for 36.854ms on Thread[1,tid=5268,Runnable,Thread*=0xf0137800,peer=0x720ac1f0,"main"]
I/Choreographer( 5268): Skipped 32 frames!  The application may be doing too much work on its main thread.
W/_thread_exampl( 5268): JNI critical lock held for 26.686ms on Thread[1,tid=5268,Runnable,Thread*=0xf0137800,peer=0x720ac1f0,"main"]
I/_thread_exampl( 5268): Background young concurrent copying GC freed 36684(1424KB) AllocSpace objects, 0(0B) LOS objects, 57% free, 1129KB/2670KB, paused 150us total 100.942ms

**Environment**

[✓] Flutter (Channel master, v1.14.6-pre.23, on Mac OS X 10.15.3 19D76, locale
    en-US)
    • Flutter version 1.14.6-pre.23 at /opt/flutter
    • Framework revision 73275f0815 (11 days ago), 2020-01-26 13:34:04 +0100
    • Engine revision 276481c4d3
    • Dart version 2.8.0 (build 2.8.0-dev.5.0 fc3af737c7)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /opt/android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /opt/android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio 3.6 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[!] Android Studio
    • Android Studio at /Applications/Android Studio 4.0 Preview.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (6 available)
    • Android SDK built for x86 • emulator-5554                            •
      android-x86    • Android 10 (API 29) (emulator)
    • MNGPH6S01                 • 1a579db335a1bc68234ccbc7f99c7989ddfc52fa • ios
      • iOS 13.3.1
    • iPhone 11 Pro Max         • 2CE40B6C-62E2-416C-94AC-A933A027B698     • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • macOS                     • macOS                                    •
      darwin-x64     • Mac OS X 10.15.3 19D76
    • Chrome                    • chrome                                   •
      web-javascript • Google Chrome 80.0.3987.87
    • Web Server                • web-server                               •
      web-javascript • Flutter Tools

! Doctor found issues in 2 categories.

**Comment by liyuqian** [comment]
I can't reliably reproduce the error message using the sample app in either https://github.com/flutter/flutter/issues/11533#issuecomment-583112880 or https://github.com/flutter/flutter/issues/11533#issuecomment-471166865. I tried to clear the storage, uninstall the app, and the error message still wouldn't show up most of the time with my Android simulator...

@mingsai @keshava19 : how often (roughly in percentages) do you see the error message?

**Comment by kf6gpe** [comment]
I'm torn with what to do with this one --- we've clearly got multiple folks reporting it, but we're waiting for input. @mingsai @keshava19, do you have any responses to @liyuqian's query on Feb 19? Or would anyone else like to chime in? I'm repplying the `waiting for customer response` label, although the bot didn't close this, to see if we get some feedback. Thanks!

**Comment by bounty1342** [comment]
I stumble upon this one. Hope this will help.

`I/Choreographer(20834): Skipped 43 frames!  The application may be doing too much work on its main thread.`

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.3 19D76, locale en)
    • Flutter version 1.12.13+hotfix.9 at /Users/bounty1342/Documents/flutter
    • Framework revision f139b11009 (8 days ago), 2020-03-30 13:57:30 -0700
    • Engine revision af51afceb8
    • Dart version 2.7.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/bounty1342/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[✓] Connected device (3 available)
    • SHIELD Android TV         • 192.168.0.82:5555 • android-arm64 • Android 9 (API 28)
    • Android SDK built for x86 • emulator-5554     • android-x86   • Android 10 (API 29) (emulator)
    • sdk google atv x86        • emulator-5556     • android-x86   • Android Q (API 28) (emulator)

• No issues found!
```

Running the app mainly in debug, can't say the frequency, but it's there.
Regards

**Comment by kf6gpe** [comment]
Thanks, @bounty1342! We'll keep this open for the time being, then, although we presently don't have anyone available to work on it.

**Comment by Hixie** [comment]
Overenthusiastic bot...

**Comment by egorges** [comment]
I also receive this frequently on emulator, Pixel 3a, API 28.

Launching lib/main.dart on AOSP on IA Emulator in debug mode...
✓ Built build/app/outputs/apk/debug/app-debug.apk.
D/FlutterActivityAndFragmentDelegate(22491): Setting up FlutterEngine.
D/FlutterActivityAndFragmentDelegate(22491): No preferred FlutterEngine was provided. Creating a new FlutterEngine for this FlutterFragment.
D/FlutterActivityAndFragmentDelegate(22491): Attaching FlutterEngine to the Activity that owns this Fragment.
D/FlutterView(22491): Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@39d128a
D/FlutterActivityAndFragmentDelegate(22491): Executing Dart entrypoint: main, and sending initial route: /
I/es.edu.sabeiss(22491): Background concurrent copying GC freed 26663(1344KB) AllocSpace objects, 12(832KB) LOS objects, 49% free, 2MB/4MB, paused 19.845ms total 375.779ms
I/Choreographer(22491): **Skipped 531 frames!**  The application may be doing too much work on its main thread.
D/EGL_emulation(22491): eglMakeCurrent: 0xe57041e0: ver 3 0 (tinfo 0xe57038c0)
D/eglCodecCommon(22491): setVertexArrayObject: set vao to 0 (0) 3 2
I/OpenGLRenderer(22491): Davey! duration=9368ms; Flags=1, IntendedVsync=52653150215684, Vsync=52662000215330, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=52662011015684, AnimationStart=52662011517684, PerformTraversalsStart=52662019948684, DrawStart=52662112528684, SyncQueued=52662125573684, SyncStart=52662143490684, IssueDrawCommandsStart=52662188812684, SwapBuffers=52662384061684, FrameCompleted=52662536966684, DequeueBufferDuration=99105000, QueueBufferDuration=14470000,

**In this thread https://stackoverflow.com/a/21126690/11252673 it is probably caused by a long initialization in the App.** 

In my case, I have a half dozen  Firebase reads. Besides being cached, it's my main point of search.

**I actually don't know if this only causes a slow start or if the App may crash on some devices. Any point on this?**

**flutter doctor -v**
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale pt-BR)
    • Flutter version 1.12.13+hotfix.8 at /Users/egr/OneDrive/Games/flutter
    • Framework revision 0b8abb4724 (3 months ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/egr/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (2 available)
    • AOSP on IA Emulator • emulator-5554                        • android-x86 • Android 9 (API 28) (emulator)
    • iPhone 8            • A3C4AFEB-ECA1-466C-A6E3-01B8CD4DB434 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-13-4
      (simulator)

• No issues found!

**Comment by larssn** [comment]
We see this a lot using the `cloud_firestore` plugin. We load at lot of data when the app starts, which obviously uses the UI thread because it's the only thread we got. It mostly happens at startup, but it can happen if you request a lot of data from Firestore - the UI locks up because the thread is now fetching data instead of rendering.

Not sure issues like these are avoidable tbh, it's just the unfortunate downside of Dart being single threaded.

**Comment by liyuqian** [comment]
@larssn : https://api.flutter.dev/flutter/foundation/compute.html. can be used to create a new thread for some heavy work.

**Comment by larssn** [comment]
@liyuqian I don't think that would help, as the native Android library is already doing a lot of the work. But the work done there still needs to be deserialized into something Flutter can use, and there's only 1 thread that can do that.

IIRC, isolates don't share memory with the main dart thread, so deserialization is unavoidable.

But I'd love to be wrong on that!

**Comment by liyuqian** [comment]
While Dart isolates don't share memory, they can send messages among each other. For your specific case, could you use https://api.flutter.dev/flutter/foundation/compute.html and pass the serialized data as Q, and return the deserialized data as R? It probably won't help much if memory I/O is the bottleneck. But if the deserialization computation is the bottleneck, I think it could help.

**Comment by pedromassangoncode** [comment]
Reproducible in `1.20.0-7.1.pre`.

<details>
<summary> flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 1.20.0-7.1.pre, on Mac OS X 10.15.5 19F101, locale en-AO)
    • Flutter version 1.20.0-7.1.pre at /Users/pedro/dev/SDKs/flutter_dev
    • Framework revision 7736f3bc90 (5 days ago), 2020-07-10 16:33:05 -0700
    • Engine revision d48085141c
    • Dart version 2.9.0 (build 2.9.0-21.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/pedro/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Users/pedro/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6514223/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Google Chrome.app/Contents/MacOS/google-chrome-unsafe

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/pedro/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6514223/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

 
[✓] Connected device (4 available)            
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86    • Android 9 (API 28) (emulator)
    • macOS (desktop)              • macos         • darwin-x64     • Mac OS X 10.15.5 19F101
    • Web Server (web)             • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 83.0.4103.116

• No issues found!
```
</details>

**Comment by techforGod** [comment]
same problem, im new to flutter/dart, makes me just want to stick to native development so I can start working on my MVP, also getting memory and gradle issues from time to time, its hella annoying

**Comment by amanv8060** [comment]
maybe this happens more when we use cloud firestore plugin  , because i often found that happening in projects which this plugin is used . this might be because cloud firestore takes substantial time to get data  if connections is slow, and while that is happenig the other process are waiting for that process to finish , which lets android consider that too much work is being done , can that be a case?

**Comment by eyoeldefare** [comment]
> maybe this happens more when we use cloud firestore plugin , because i often found that happening in projects which this plugin is used . this might be because cloud firestore takes substantial time to get data if connections is slow, and while that is happenig the other process are waiting for that process to finish , which lets android consider that too much work is being done , can that be a case?

It doesn't have to be Firestore, anything that puts tons of workload on the main thread is going to show this warning. I am guessing it has to do with dart being a single thread wherein other multithread applications, you could probably create more threads when one thread is overwhelmed to handle tasks. This is the weakness of asynchronous programming. When you keep adding too many tasks, the time it takes to do those added tasks just gets bigger and bigger until it just abandons it in this case by jumping frames to help alleviate resources.

**Comment by liyuqian** [comment]
@eyoeldefare : https://api.flutter.dev/flutter/foundation/compute.html can be used to create a new thread/isolate in Dart for concurrent computations.

**Comment by kf6gpe** [comment]
I'm fuzzy about actionable work here --- it seems like there's been some good discussion of alternatives, but the threading mode is part of how the Dart VM works. Are there specific use cases like cloud_firestore that we can open separate bugs against? (And is cloud_firestore one that I should do so with?)

**Comment by liyuqian** [comment]
@kf6gpe : recently, we found that this issue is mainly about blocking the main thread for too long during engine startup. There are several attempts to fix it (see, e.g., flutter/engine#18047, flutter/engine#20142, flutter/engine#20433). We'll figure out how many resources we have in Q4 to properly address it.

**Comment by RashidaBegamShaik** [comment]
am facing this issue when the app is killed/removed from background and when I get notification, I clicked on notification ... then its on navigation it should navigate to particular screen , its not happening.. just opened splash screen and in the console this message was occurred

**Comment by andicooxy** [comment]
I'm also here for solutions, 
`Skipped 51 frames!  The application may be doing too much work on its main thread.
D/InputMethodManager(13830): prepareNavigationBarInfo() DecorView@60d01c4[MainActivity]
D/InputMethodManager(13830): getNavigationBarColor() -855310
V/InputMethodManager(13830): Starting input: tba=com.example.realityApp ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
D/InputMethodManager(13830): startInputInner - Id : 0
I/InputMethodManager(13830): startInputInner - mService.startInputOrWindowGainedFocus
D/InputTransport(13830): Input channel destroyed: ‘ClientS’, fd=152
D/InputMethodManager(13830): prepareNavigationBarInfo() DecorView@60d01c4[MainActivity]
D/InputMethodManager(13830): getNavigationBarColor() -855310
V/InputMethodManager(13830): Starting input: tba=com.example.realityApp ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false`

**Comment by andicooxy** [comment]
> I'm also here for solutions,
> `Skipped 51 frames! The application may be doing too much work on its main thread. D/InputMethodManager(13830): prepareNavigationBarInfo() DecorView@60d01c4[MainActivity] D/InputMethodManager(13830): getNavigationBarColor() -855310 V/InputMethodManager(13830): Starting input: tba=com.example.realityApp ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false D/InputMethodManager(13830): startInputInner - Id : 0 I/InputMethodManager(13830): startInputInner - mService.startInputOrWindowGainedFocus D/InputTransport(13830): Input channel destroyed: ‘ClientS’, fd=152 D/InputMethodManager(13830): prepareNavigationBarInfo() DecorView@60d01c4[MainActivity] D/InputMethodManager(13830): getNavigationBarColor() -855310 V/InputMethodManager(13830): Starting input: tba=com.example.realityApp ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false`

My issue was resolved by reducing image sizes down to 50 kb

**Comment by thedevsbuddy** [comment]
I am getting this message even in a fresh project I mean the default project which flutter gives us when we create a new project.

**Comment by AndreyKalashnik** [comment]
Make sure you have not put a **Breakpoint** in your _Code editor_. I missed how I put it, but spent extra hour of being confusing 😵

**Comment by darshankawar** [comment]
I verified the originally reported issue using latest master version and by creating fresh Flutter project and ran it on Android emulator which didn't give me the said log / message.

```
dhs@Dhss-MacBook-Pro enable_settings_issue % flutter run
Multiple devices found:
Android SDK built for x86 (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
iPhone 13 (mobile)                 • 3BD8F2FF-FDB5-4074-8C91-54F99C79156D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
macOS (desktop)                    • macos                                • darwin-x64     • macOS 12.2.1 21D62 darwin-x64
Chrome (web)                       • chrome                               • web-javascript • Google Chrome 111.0.5563.64
[1]: Android SDK built for x86 (emulator-5554)
[2]: iPhone 13 (3BD8F2FF-FDB5-4074-8C91-54F99C79156D)
[3]: macOS (macos)
[4]: Chrome (chrome)
Please choose one (or "q" to quit): 1
Using hardware rendering with device Android SDK built for x86. If you notice graphics artifacts, consider enabling software rendering with "--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Running Gradle task 'assembleDebug'...                             32.3s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...        1,186ms
Syncing files to device Android SDK built for x86...               125ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

A Dart VM Service on Android SDK built for x86 is available at: http://127.0.0.1:55580/778WOhLX32M=/
The Flutter DevTools debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:9102?uri=http://127.0.0.1:55580/778WOhLX32M=/


```

`Channel master, 3.9.0-1.0.pre.183`

@adityamv7 Can you confirm if this is still an issue at your end using latest versions of the framework ?
